Mage.Sets/src/mage/cards/a/AnnihilatingGlare.java

```diff
@@ -0,0 +1,47 @@
+package mage.cards.a;
+
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.OrCost;
+import mage.abilities.costs.common.SacrificeTargetCost;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.common.DestroyTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.StaticFilters;
+import mage.target.common.TargetControlledPermanent;
+import mage.target.common.TargetCreatureOrPlaneswalker;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author AhmadYProjects
+ */
+public final class AnnihilatingGlare extends CardImpl {
+
+    public AnnihilatingGlare(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{B}");
+        
+
+        // As an additional cost to cast this spell, pay {4} or sacrifice an artifact or creature.
+        final Cost manaCost = new GenericManaCost(4);
+        final Cost sacrificeCost = new SacrificeTargetCost(new TargetControlledPermanent(StaticFilters.FILTER_CONTROLLED_PERMANENT_ARTIFACT_OR_CREATURE));
```

Comment:
these costs don't need to be final and you don't need to declare them separately either

Reply:
they weren't declared seperately initially but i was having issues when testing the card so i tried to seperate them. I have since fixed it and implement requested changes and card works as it should